src/libraries/System.Runtime.Serialization.BinaryFormat/src/System/Runtime/Serialization/BinaryFormat/RecordType.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+/// <summary>
+/// Record type.
+/// </summary>
+/// <remarks>
+///  <para>
+///   The enumeration does not contain all values supported by the <see href="https://learn.microsoft.com/openspecs/windows_protocols/ms-nrbf/954a0657-b901-4813-9398-4ec732fe8b32">
+///   [MS-NRBF] 2.1.2.1</see>, but only those supported by the <see cref="PayloadReader"/>.
+///  </para>
+/// </remarks>
+#if SYSTEM_RUNTIME_SERIALIZATION_BINARYFORMAT
+public
+#else
+internal
+#endif
+enum RecordType : byte
+{
+    /// <summary>
+    /// The NRBF header (the first record in NRBF payload).
+    /// </summary>
+    SerializedStreamHeader = 0,
+    /// <summary>
+    /// A class information that references another class record's metadata.
+    /// </summary>
+    ClassWithId = 1,
+    // SystemClassWithMembers and ClassWithMembers are not supported by design (require type loading)
+    /// <summary>
+    /// A system class information with type info.
+    /// </summary>
+    SystemClassWithMembersAndTypes = 4,
+    /// <summary>
+    /// A class information with type info and the source library.
+    /// </summary>
+    ClassWithMembersAndTypes = 5,
+    /// <summary>
+    /// A <see langword="string" />.
+    /// </summary>
+    BinaryObjectString = 6,
+    /// <summary>
+    /// An array other than single dimensional array of primitive types and <see cref="object" />.
+    /// </summary>
+    BinaryArray = 7,
+    /// <summary>
+    /// A primitive value other than <see langword="string"/>.
+    /// </summary>
+    MemberPrimitiveTyped = 8,
+    /// <summary>
+    /// A record that contains a reference to another record that contains the actual value.
+    /// </summary>
+    MemberReference = 9,
+    /// <summary>
+    /// A <see langword="null" />.
+    /// </summary>
+    ObjectNull = 10,
+    /// <summary>
+    /// The record that marks the end of the binary format stream.
+    /// </summary>
+    MessageEnd = 11,
+    /// <summary>
+    /// A library is a named unit that contains a collection of types.
```

Comment:
That's a really weird doc comment.

```suggestion
    /// A record that associates a numeric identifier with a named library.
```

Reply:
I know, but to my defense I've copy-pasted the description from https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/8b313786-0baf-4f01-bc45-3a4c70af3e01#gt_ba47762a-6279-48b3-9a11-eeb239876e31:

> Library: Part of the [Remoting Data Model](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/8b313786-0baf-4f01-bc45-3a4c70af3e01#gt_326c1f46-aafe-4f44-b312-ba017c2baaf0). A [Library](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/8b313786-0baf-4f01-bc45-3a4c70af3e01#gt_ba47762a-6279-48b3-9a11-eeb239876e31) is a named unit that contains a collection of [Remoting Types](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/8b313786-0baf-4f01-bc45-3a4c70af3e01#gt_13079b7f-eef0-408e-a789-71f204557586). For more information, see Library in [MS-NRTP] section 3.1.1.